File: backend/src/main/java/com/example/blog/entity/Comment.java
package com.example.blog.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "comments")
@Getter
@Setter
public class Comment extends BaseEntity {

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Column(length = 50)
    private String authorName;

    @Column(length = 100)
    private String authorEmail;

    @Column(length = 255)
    private String authorUrl;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Status status = Status.PENDING;

    @Column(length = 50)
    private String ipAddress;

    @Column(length = 255)
    private String userAgent;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "article_id", nullable = false)
    private Article article;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id")
    private Comment parent;

    @OneToMany(mappedBy = "parent", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Comment> replies = new ArrayList<>();

    public enum Status {
        PENDING, APPROVED, REJECTED
    }
}

File: frontend/packages/shared/src/types/comment.ts
export type CommentStatus = 'PENDING' | 'APPROVED' | 'REJECTED'

export interface CommentDTO {
  id: number
  content: string
  authorName: string | null
  authorEmail: string | null
  authorUrl: string | null
  status: CommentStatus
  articleId: number
  articleTitle: string
  parentId: number | null
  replies: CommentDTO[]
  createdAt: string
  updatedAt: string
}

export interface CommentRequest {
  content: string
  authorName?: string
  authorEmail?: string
  authorUrl?: string
  parentId?: number
}
